kroxylicious-additional-filters/kroxylicious-multitenant/src/main/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilter.java

```diff
@@ -145,6 +145,8 @@ public void onCreateTopicsResponse(short apiVersion, ResponseHeaderData header,
 
     @Override
     public void onDeleteTopicsRequest(short apiVersion, RequestHeaderData header, DeleteTopicsRequestData request, KrpcFilterContext context) {
+        // the topicName field was present up to and including version 5
+        request.setTopicNames(request.topicNames().stream().map(topic -> applyTenantPrefix(context, topic)).toList());
```

Comment:
should this be behind a guard condition of `apiVersion <= 5`?

Looks like its safe as topicNames is initialised to an empty list, but I wonder if a guard condition would be better than a comment for conveying the behaviour.

Reply:
I'm not sure.  On the one hand yes, comments rot, code less so.  OTOH, the Kafka Client Library is doing a good job of making sure older fields are initialised in such a way to avoid NPEs, maybe keeping the execution path straight is better.

I'm think we defer making a decision until we learn a bit more about writing filters that work across versions.